src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts

```diff
@@ -105,6 +105,11 @@ export class CrushmapComponent implements OnInit {
   }
 
   onUpdateData() {
+    setTimeout(() => {
+      this.healthService.getFullHealth().subscribe((data: any) => {
```

Comment:
Tested and it works, but this implementation is tightly coupled to onUpdateData calls which may change and we're repeating the same call that we have in ngOnInit.
When leaving the page, a last call is still made. To avoid this,
we already have a TimerService for polling so we could use it, sth like this (example):
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts
index af8b83f8c66..0e62c17c5de 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts
@@ -1,4 +1,4 @@
-import { Component, OnInit, ViewChild } from '@angular/core';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
 
 import {
   ITreeOptions,
@@ -7,8 +7,10 @@ import {
   TreeNode,
   TREE_ACTIONS
 } from '@circlon/angular-tree-component';
+import { Subscription } from 'rxjs';
 
 import { HealthService } from '~/app/shared/api/health.service';
+import { TimerService } from '~/app/shared/services/timer.service';
 import { Icons } from '~/app/shared/enum/icons.enum';
 
 @Component({
@@ -16,7 +18,9 @@ import { Icons } from '~/app/shared/enum/icons.enum';
   templateUrl: './crushmap.component.html',
   styleUrls: ['./crushmap.component.scss']
 })
-export class CrushmapComponent implements OnInit {
+export class CrushmapComponent implements OnInit, OnDestroy {
+  private sub = new Subscription();
+
   @ViewChild('tree') tree: TreeComponent;
 
   icons = Icons;
@@ -36,13 +40,20 @@ export class CrushmapComponent implements OnInit {
   metadataTitle: string;
   metadataKeyMap: { [key: number]: any } = {};
 
-  constructor(private healthService: HealthService) {}
+  constructor(private healthService: HealthService, private timerService: TimerService) {}
 
   ngOnInit() {
-    this.healthService.getFullHealth().subscribe((data: any) => {
-      this.loadingIndicator = false;
-      this.nodes = this.abstractTreeData(data);
-    });
+    this.sub = this.timerService
+      .get(() => this.healthService.getFullHealth(), 5000)
+      .subscribe((data: any) => {
+        console.log('ON UPDATE');
+        this.loadingIndicator = false;
+        this.nodes = this.abstractTreeData(data);
+      });
+  }
+
+  ngOnDestroy() {
+    this.sub.unsubscribe();
   }
 
   private abstractTreeData(data: any): any[] {
@@ -105,11 +116,6 @@ export class CrushmapComponent implements OnInit {
   }
 
   onUpdateData() {
-    setTimeout(() => {
-      this.healthService.getFullHealth().subscribe((data: any) => {
-        this.nodes = this.abstractTreeData(data);
-      });
-    }, 5000);
     this.tree.treeModel.expandAll();
   }
 }
```

Reply:
> Tested and it works, but this implementation is tightly coupled to onUpdateData calls which may change and we're repeating the same call that we have in ngOnInit.
> When leaving the page, a last call is still made. To avoid this,
> we already have a TimerService for polling so we could use it, sth like this (example):
> 
> ```
> diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts
> index af8b83f8c66..0e62c17c5de 100644
> --- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts
> +++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts
> @@ -1,4 +1,4 @@
> -import { Component, OnInit, ViewChild } from '@angular/core';
> +import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
>  
>  import {
>    ITreeOptions,
> @@ -7,8 +7,10 @@ import {
>    TreeNode,
>    TREE_ACTIONS
>  } from '@circlon/angular-tree-component';
> +import { Subscription } from 'rxjs';
>  
>  import { HealthService } from '~/app/shared/api/health.service';
> +import { TimerService } from '~/app/shared/services/timer.service';
>  import { Icons } from '~/app/shared/enum/icons.enum';
>  
>  @Component({
> @@ -16,7 +18,9 @@ import { Icons } from '~/app/shared/enum/icons.enum';
>    templateUrl: './crushmap.component.html',
>    styleUrls: ['./crushmap.component.scss']
>  })
> -export class CrushmapComponent implements OnInit {
> +export class CrushmapComponent implements OnInit, OnDestroy {
> +  private sub = new Subscription();
> +
>    @ViewChild('tree') tree: TreeComponent;
>  
>    icons = Icons;
> @@ -36,13 +40,20 @@ export class CrushmapComponent implements OnInit {
>    metadataTitle: string;
>    metadataKeyMap: { [key: number]: any } = {};
>  
> -  constructor(private healthService: HealthService) {}
> +  constructor(private healthService: HealthService, private timerService: TimerService) {}
>  
>    ngOnInit() {
> -    this.healthService.getFullHealth().subscribe((data: any) => {
> -      this.loadingIndicator = false;
> -      this.nodes = this.abstractTreeData(data);
> -    });
> +    this.sub = this.timerService
> +      .get(() => this.healthService.getFullHealth(), 5000)
> +      .subscribe((data: any) => {
> +        this.loadingIndicator = false;
> +        this.nodes = this.abstractTreeData(data);
> +      });
> +  }
> +
> +  ngOnDestroy() {
> +    this.sub.unsubscribe();
>    }
>  
>    private abstractTreeData(data: any): any[] {
> @@ -105,11 +116,6 @@ export class CrushmapComponent implements OnInit {
>    }
>  
>    onUpdateData() {
> -    setTimeout(() => {
> -      this.healthService.getFullHealth().subscribe((data: any) => {
> -        this.nodes = this.abstractTreeData(data);
> -      });
> -    }, 5000);
>      this.tree.treeModel.expandAll();
>    }
>  }
> ```

Thanks @alfonsomthd for the suggesstion! I've updated the PR with these changes.